### ISSUE TITLE
shape of y_batch of the generator vs the output of the model doesn't match

### ISSUE THREAD

**Issue by sadakmed** [snakers4/yolov2-fish#1]
Hii mate thank u for sharing this implementation it's really nice as a beginner.

during my cracking to your code, I didn't get it, normally u output the model as one-hot encoded but in y_batch of the generator is label encoded

this is get_generator from BatchGenerator class
`        y_batch = np.zeros((self.config['BATCH_SIZE'], self.config['GRID_H'], self.config['GRID_W'], self.config['BOX'], 4+1+1))                # desired network output
` 
this is the output of the model 

`
output = Reshape((GRID_H, GRID_W, BOX, 4 + 1 + CLASS))(x)
`

what do you think?

**Comment by snakers4** [comment]
oh, this repo is such a mess
better just take some clean implementation and train it from scratch )

**Comment by sadakmed** [comment]
@snakers4  could you please recommend me something

**Comment by snakers4** [comment]
https://github.com/eriklindernoren/PyTorch-YOLOv3
